### PR TITLE
Add Makefile and use fixed linter

### PR DESCRIPTION
**Pull request by sebastienvas**
```release-note
none
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+test:
+	@bazel test //...
+
+lint: build
+	@bin/linters.sh
+	@bin/headers.sh
+
+coverage:
+	@bin/codecov.sh
```
Comment (lookuptable) [inline Makefile]: coverage depends on setup I believe

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+clean:
+	@bazel clean
+
+test:
+	@bazel test //...
+
+lint: build
+	@bin/linters.sh
+	@bin/headers.sh
```
Comment (lookuptable) [inline Makefile]: It might make more sense to create a separate target for headers but this is really a nit

Comment (sebastienvas) [reply Makefile]: I am just following mixer's one. Such that all makefile behave the same.

File: Makefile (hunk 1 above)
Comment (sebastienvas) [reply Makefile]: Done

File: bin/linters.sh [hunk 3]
```diff
@@ -1,8 +1,15 @@
...
-gometalinter --concurrency=4 --enable-gc --deadline=300s --disable-all\
-  --enable=maligned\
+LINTER_SHA='bfcc1d6942136fd86eb6f1a6fb328de8398fbd80'
+
+docker run \
+  -v $(bazel info output_base):$(bazel info output_base) \
+  -v $(pwd):/go/src/istio.io/auth \
+  -w /go/src/istio.io/auth gcr.io/istio-testing/linter:${LINTER_SHA} \
+  --concurrency=4 --enable-gc --deadline=300s --disable-all\
+  --enable=aligncheck\
```
Comment (myidpt) [inline bin/linters.sh]: I think aligncheck is deprecated in favor of maligned:
https://github.com/alecthomas/gometalinter/pull/367

File: prow/auth-presubmit.sh [hunk 4]
```diff
@@ -40,7 +40,6 @@ else
 fi
 
 echo "=== Bazel Build ==="
-./bin/install-prereqs.sh
```
Comment (myidpt) [inline prow/auth-presubmit.sh]: Is it always true that we don't need it?

File: bin/linters.sh (hunk 3 above)
Comment (sebastienvas) [reply bin/linters.sh]: Yes but the version that we use in the docker image is older than this.

File: prow/auth-presubmit.sh (hunk 4 above)
Comment (sebastienvas) [reply prow/auth-presubmit.sh]: this just install gometalinter which is not needed.

**Pull request merged**